sapi/phpdbg/phpdbg_lexer.l

```diff
@@ -84,7 +84,10 @@ ADDR        [0][x][a-fA-F0-9]+
 OPCODE      (ZEND_|zend_)([A-Za-z])+
 INPUT       ("\\"[#"']|["]("\\\\"|"\\"["]|[^\n\000"])+["]|[']("\\"[']|"\\\\"|[^\n\000'])+[']|[^\n\000#"'])+
 
-<!*> := yyleng = (size_t) YYCURSOR - (size_t) yytext;
+<!*> {
+	if (YYCURSOR == NULL) return T_UNEXPECTED;
```

Comment:
How does YYCURSOR become null?

Reply:
Condider the following input: `"a\b"`. When backtracking, re2c sets `YYCURSOR` to `YYMARKER`, which is `NULL`. The [re2c documentation](https://re2c.org/manual/manual_c.html#api-primitives) says it would not be necessary to initialize YYMARKER, but when I [do it](https://github.com/php/php-src/blob/php-7.4.13/sapi/phpdbg/phpdbg_lexer.l#L36) nonetheless, there is no NULL issue here anymore, but for some reason the lexer enters the `<NORMAL>{GENERIC_ID}` rule, albeit with an empty text (`yyleng == 0`) what causes issues in `unescape_string()`. I don't understand why that rule is executed, since `GENERIC_ID` is not supposed to be empty. Anyway, if I add a suitable default rule `<NORMAL>* {return T_UNEXPECTED;}`, I get the desired parse error.